src/collapse/perf.rs

```diff
@@ -154,29 +188,96 @@ impl Collapse for Folder {
     }
 }
 
-impl From<Options> for Folder {
-    fn from(mut opt: Options) -> Self {
-        opt.include_pid = opt.include_pid || opt.include_tid;
-        Self {
-            cache_line: Vec::default(),
-            event_filtering: EventFilterState::None,
-            in_event: false,
-            occurrences: FnvHashMap::with_capacity_and_hasher(512, fnv::FnvBuildHasher::default()),
-            pname: String::new(),
-            skip_stack: false,
-            stack: VecDeque::default(),
-            opt,
+impl Folder {
+    fn collapse_single_threaded<R>(&mut self, mut reader: R) -> io::Result<()>
+    where
+        R: BufRead,
+    {
+        let mut line = String::new();
+        loop {
+            line.clear();
+            if reader.read_line(&mut line)? == 0 {
+                break;
+            }
+            if line.starts_with('#') {
+                continue;
+            }
+            let line = line.trim_end();
+            if line.is_empty() {
+                self.after_event();
+            } else if self.in_event {
+                self.on_stack_line(line)
+            } else {
+                self.on_event_line(line)
+            }
         }
+        Ok(())
```

Comment:
Could probably get rid of this by making the `break` in the `loop` above be `break Ok(())`.

Reply:
Come to think of it, that likely also applies to the dtrace collapser.

Reply:
Haha, you don't appear to like the `Ok(())` at the bottom of functions much. I actually kind of like it, but of course will defer to your aesthetic judgement :)

Reply:
Haha, I mean, its not super important to me, I just don't like to pollute code with unnecessary statements. If you have a strong preference one way, I'm happy with that :)

Reply:
No. I'll change it. I don't have a strong preference either way either. :)